File: frontend/services/resumes.ts
import { apiRequest } from "@/services/api-client";

export interface ResumeProcessResult {
  attachments_scanned: number;
  applicants_created: number;
  needs_review: number;
  failed: number;
  acknowledgments_sent: number;
  acknowledgment_errors: number;
}

interface ResumeProcessResponse {
  success: boolean;
  message: string;
  data: ResumeProcessResult;
}

export async function processStoredResumes(): Promise<ResumeProcessResult> {
  const response = await apiRequest<ResumeProcessResponse>("/resumes/process", {
    method: "POST",
    body: JSON.stringify({ max_attachments: 25 }),
  });
  return response.data;
}

File: backend/app/schemas/resumes.py
from pydantic import BaseModel, Field


class ResumeProcessRequest(BaseModel):
    max_attachments: int = Field(default=25, ge=1, le=100)


class ResumeProcessResult(BaseModel):
    attachments_scanned: int = 0
    applicants_created: int = 0
    needs_review: int = 0
    failed: int = 0
    acknowledgments_sent: int = 0
    acknowledgment_errors: int = 0


class ResumeProcessResponse(BaseModel):
    success: bool = True
    message: str
    data: ResumeProcessResult
